homeassistant/components/habitica/__init__.py

```diff
@@ -122,6 +143,73 @@ class HAHabitipyAsync(HabitipyAsync):
         def __call__(self, **kwargs):
             return super().__call__(websession, **kwargs)
 
+    async def cast_skill(call: ServiceCall) -> ServiceResponse:
+        """Skill action."""
+        entry: HabiticaConfigEntry | None
+        if entry := hass.config_entries.async_get_entry(call.data[ATTR_CONFIG_ENTRY]):
+            coordinator = entry.runtime_data
+            skill = {
+                "pickpocket": {"spellId": "pickPocket", "cost": "10 MP"},
+                "backstab": {"spellId": "backStab", "cost": "15 MP"},
+                "smash": {"spellId": "smash", "cost": "10 MP"},
+                "fireball": {"spellId": "fireball", "cost": "10 MP"},
+            }
+            try:
+                task_id = next(
+                    task["id"]
+                    for task in coordinator.data.tasks
+                    if call.data[ATTR_TASK] in (task["id"], task.get("alias"))
+                    or call.data[ATTR_TASK] == task["text"]
+                )
+            except StopIteration as e:
+                raise ServiceValidationError(
+                    translation_domain=DOMAIN,
+                    translation_key="service_call_task_not_found",
+                    translation_placeholders={"task": f"'{call.data[ATTR_TASK]}'"},
+                ) from e
+
+            try:
+                response: dict[str, Any] = await coordinator.api.user.class_.cast[
+                    skill[call.data[ATTR_SKILL]]["spellId"]
+                ].post(targetId=task_id)
+            except ClientResponseError as e:
+                if e.status == HTTPStatus.TOO_MANY_REQUESTS:
+                    raise ServiceValidationError(
+                        translation_domain=DOMAIN,
+                        translation_key="setup_rate_limit_exception",
+                    ) from e
+                if e.status == HTTPStatus.UNAUTHORIZED:
+                    raise ServiceValidationError(
+                        translation_domain=DOMAIN,
+                        translation_key="service_call_not_enough_mana",
+                        translation_placeholders={
+                            "cost": skill[call.data[ATTR_SKILL]]["cost"],
+                            "mana": f"{int(coordinator.data.user.get("stats", {}).get("mp", 0))} MP",
```

Comment:
Why do we `int` the response? It will be stringified later anyway

Reply:
This is a float value, I wanted to cut off decimal places.

Reply:
Oh, hmm, would it make sense to use `round`?

Reply:
Let's say we have 9.9 Mana, and the spell costs 10, the error message would read you need 10 Mana but you only have 10. I think cutting off decimals and displaying 9 is better than rounding.